### PR TITLE
Adsk update lightbindings

### PR DESCRIPTION
**Pull request by niklasharrysson**

### CODE REVIEW

File: source/MaterialXTest/ShaderValid.cpp [hunk 1]
```diff
@@ -351,33 +352,18 @@ TEST_CASE("GLSL Validation from HwShader", "[shadervalid]")
         const std::string lightDoc = " \
         <?xml version=\"1.0\"?> \
         <materialx version=\"1.35\" require=\"\"> \
-          <nodegraph name=\"lighting1\" type=\"\" xpos=\"10.9034\" ypos=\"15.42\" adskDisplayMode=\"1\"> \
```
Comment (niklasharrysson) [inline source/MaterialXTest/ShaderValid.cpp]: Simplified the test shader to a single white diffuse BSDF, to see the lighting effects better.

File: source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp [hunk 2]
```diff
@@ -389,97 +397,117 @@ void GlslValidator::bindLighting()
...
+            }
         }
+
+        ++index;
     }
-    Input = uniformList.find("u_lightData[0].color");
-    if (Input != uniformList.end())
+}
+
+void GlslValidator::setUniform(int location, const Value& value)
```
Comment (niklasharrysson) [inline source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp]: Utility method to set uniform from a Value

File: source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp [hunk 3]
```diff
@@ -376,6 +376,14 @@ void GlslValidator::bindTimeAndFrame()
 
 void GlslValidator::bindLighting()
 {
+    if (!_lightHandler)
```
Comment (niklasharrysson) [inline source/MaterialXView/ShaderValidators/Glsl/GlslValidator.cpp]: Binding light data from the light handler.

**Comment by bernardkwok** [comment]
This looks great. Thanks for adding in the uniform setting utility.

**Comment by bernardkwok** [comment]
Sorry for closing by accident. Was trying to approve.

**Pull request merged**